### ISSUE TITLE
Does it `sanitize` reserved name like CON or COM1 on Windows?

### ISSUE THREAD

**Issue by vi** [AndyBarron/app-dirs-rs#14]
Clicking at `[src]` in the documentation does not show any special handling of things like

    CON, PRN, AUX, NUL, COM1, COM2, COM3, COM4, COM5, COM6, COM7, COM8, COM9, LPT1, LPT2, LPT3, LPT4, LPT5, LPT6, LPT7, LPT8, LPT9

([source](http://superuser.com/a/844514/27264))

Is it because of online documentation isn't for windows or it is not handled?

**Comment by AndyBarron** [comment]
Not currently handled, but would be a great (and simple) addition.

[It looks like any name *starting* with those values and a period are also invalid.](https://social.technet.microsoft.com/Forums/windows/en-US/e22c021d-d188-4ff2-a4dd-b5d58d979c58/the-specified-device-name-is-invalid)

**Comment by vi** [comment]
If `sanitized` is indented to be used for security, this would be a necessary addition.

It should probably stated in the documentation whether the function is expected to be secure or not.

**Comment by AndyBarron** [comment]
The intent wasn't security, but to make sure that e.g. somebody developing on Linux wouldn't accidentally use path names that fail on Windows.

We call `sanitized` on every platform for the sake of consistency, but I'll probably handled the "reserved" names on Windows only (to avoid a breaking change, since those names would fail on Windows anyway).

**Comment by vi** [comment]
> The intent wasn't security

Then either the scope and intent should be enlarged to include uses for security (e.g. the name is 100% guranteed to be "no tricks"), or phrase like "the function should not be used for security, to sanitize hostile user input" or "it is possible that `sanitized` returns a tricky filename on some platforms".

**Comment by AndyBarron** [comment]
Agreed; I can add a clarification to the docs. It definitely isn't intended to be used on *user* input, but programmer input - to create a directory structure for their app data in the right place.

**Comment by vi** [comment]
A secure filename sanitizer may worth a separate crate. It may be full of heuristics and tweaks (like "no unicode + abbreviate + supplement with a short hash" or "please make a 8.3 filename")
